### PR TITLE
Issue #2930 . Handeling exceptions during GetProvider to prevent IDE crash

### PR DESCRIPTION
**Pull request by jascam**

### CODE REVIEW

File: src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs [hunk 1]
```diff
@@ -84,29 +84,47 @@ public class VsCredentialProviderImporter
...
         {
-            this._initializer();
             ICredentialProvider result = null;
 
-            if (ImportedProvider != null)
+            try
             {
-                result = new VsCredentialProviderAdapter(ImportedProvider);
-            }
+                this._initializer();
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: nit: We don't use `this.` in this codebase

File: src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs [hunk 2]
```diff
@@ -84,29 +84,47 @@ public class VsCredentialProviderImporter
...
                 }
             }
+            catch (Exception e)
+            {
+                // Typicall catching all exceptions is discuraged however the code calling this method
+                // is in the initialization of NuGet and is checking for the existance of a credential
+                // provider.   Crashing here is crashing at the NuGet startup in the IDE.  Considering
+                // the scenario is not close to 100% of the NuGet usage scenarios catching the exception
+                // writting the error and proceeding without the provider is a safer option.
+                this._errorDelegate(string.Format(Resources.VsCredentialProviderImport_GenericError,
```
Comment (yishaigalatzer) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: Where is this error going to?

**Comment by yishaigalatzer** [comment]
1. Please add a link to the issue, https://github.com/NuGet/home/issues/2419 is not exactly related. Should we have a new bug or more specific info?
2. Change looks good overall, got some minor comments and a question.
3. What release does this need to go into? Or in other words what is the urgency on this fix? We are locking down 3.4.5 by the end of the week, and it will tentatively go into U3. Being in ask mode, I'd like to hear a justification if you want it to go in quickly or wait for 3.5 (which may take a few months).

CC @rrelyea

File: src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs (hunk 1 above)
Comment (jascam) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: The code exists this way before this change.  I would prefer not to increase the changes as this was supposed to be targeted.  As this is a nit are you ok with won't fix?

Comment (yishaigalatzer) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: I'm ok with won't fix, but I would appreciate another cleanup commit that goes just into dev. If we don't cleanup as we go through code it just never gets prioritized.

Not going to push hard, will leave it up to you

**Comment by jascam** [comment]
1. I'll create a new bug and link it.
2. I'll address the comments.
3. The primary risk was addressed when 2419 was marked as closed this is more follow up work.  I don't think this needs to be rushed into 3.4.5.

File: src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs (hunk 2 above)
Comment (jascam) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: This is currently _outputConsoleLogger.OutputConsole.WriteLine

Comment (yishaigalatzer) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: We should probably write this one out to the activitylog.xml as well

File: src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs (hunk 1 above)
Comment (jascam) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: Fixed in second commit

File: src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs (hunk 2 above)
Comment (jascam) [inline src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs]: Fixed in second commit

**Comment by yishaigalatzer** [comment]
lgtm

CC @alpaix @emgarten @rrelyea

**Comment by alpaix** [comment]
Looks good. Please squash commits when merging the PR.
:shipit:

**Comment by alpaix** [comment]
Merged into dev https://github.com/NuGet/NuGet.Client/commit/4d034c32e070f02323dc0cd2e9753e774ebc74c2

**Pull request closed without merging**